Add FHIR GET requests to retrieve all resources conforming to each mCODE profile

masnick opened the pull request:
This adds a list of FHIR requests that retrieve the resources corresponding to each mCODE profile.

It is not currently possible to construct requests for some profiles due to lack of required value set bindings or fixed codes. (If there's some clever way to get around this that I missed, please let me know!)

I tested all the query params against Logica (`code:in=`, `code=`, etc.) and everything in the PR worked.

markkramerus commented:
1) This one, do you have to have both a LOINC code AND a category? Or is the code sufficient? CancerGenomicsReport: GET [base]/DiagnosticReport?code=http://loinc.org|81247-9&category=http://terminology.hl7.org/CodeSystem/v2-0074|GE

2) CancerRelatedMedicationRequest: I thought the "cancer-relatedness" came from requiring either a reasonCode or reasonReference. Can you make a query that filters out MedReqs that have a cancer-related reason? (If not, at least add an explanation).

3) Same with MedAdmin

4) GeneticSpecimen -- this would be any Specimen that is related to a Genomics test. Again, perhaps not directly query-able, but at least identifiable.

masnick commented:
Hi @markkramerus:

> 1. This one, do you have to have both a LOINC code AND a category? Or is the code sufficient? CancerGenomicsReport: GET [base]/DiagnosticReport?code=[http://loinc.org|81247-9&category=http://terminology.hl7.org/CodeSystem/v2-0074|GE](http://loinc.org%7C81247-9&category=http://terminology.hl7.org/CodeSystem/v2-0074%7CGE)

This was inconsistent with our conformance prose. I removed `category` from the request.

> 2. CancerRelatedMedicationRequest: I thought the "cancer-relatedness" came from requiring either a reasonCode or reasonReference. Can you make a query that filters out MedReqs that have a cancer-related reason? (If not, at least add an explanation).
> 3. Same with MedAdmin

I added a prose note about this. Representing computably would require multiple separate queries, which I can attempt do if you think it's worthwhile.

> 4. GeneticSpecimen -- this would be any Specimen that is related to a Genomics test. Again, perhaps not directly query-able, but at least identifiable.

I added a prose note about this. Representing computably would require multiple separate queries, which I can attempt to do if you think it's worthwhile.

----

In addition to responding to your comments, I went through and confirmed the requests match the prose conformance guidance on each profile page. As I did this, I also made a number of style and consistency fixes to `*-intro.md` files for the profiles.

I did notice that Tumor and TumorSpecimen do not have `-intro.md` pages, so there is no profile-specific conformance guidance. Do we want to fix that?
